xarray/core/computation.py

```diff
@@ -584,7 +592,7 @@ def apply_groupby_func(func, *args):
     iterators = []
     for arg in args:
         if isinstance(arg, GroupBy):
-            iterator = (value for _, value in arg)
+            iterator: Iterator = (value for _, value in arg)
```

Comment:
Two things here: usually it is better to define the type outside of the if-else.
Something like:
```python
iterator: Iterator[Any]
if ....
    iterator = ...
else:
    iterator= ...
```
And second, always add the generic type like in the example above. If you don't know it, use Any.